packages/atomic/src/components/commerce/atomic-commerce-product-list/atomic-commerce-product-list.tsx

```diff
@@ -40,22 +67,52 @@ export class AtomicCommerceProductList
   @BindStateToController('search')
   @State()
   private searchState!: SearchState;
+  @State() private resultTemplateRegistered = false;
   @State() public error!: Error;
+  @State() private templateHasError = false;
 
   /**
    * The desired layout to use when displaying products. Layouts affect how many products to display per row and how visually distinct they are from each other.
    */
-  @Prop({reflect: true}) display: 'grid' | 'list' = 'grid';
+  @Prop({reflect: true}) display: ItemDisplayLayout = 'grid';
 
   /**
    * The spacing of various elements in the product list, including the gap between products, the gap between parts of a product, and the font sizes of different parts in a product.
    */
-  @Prop({reflect: true}) density: 'normal' | 'compact' = 'normal';
+  @Prop({reflect: true}) density: ItemDisplayDensity = 'normal';
 
   /**
    * The expected size of the image displayed for products.
    */
-  @Prop({reflect: true}) imageSize: number = 400;
+  @Prop({reflect: true}) imageSize: ItemDisplayImageSize = 'small';
+
+  /**
+   * The target location to open the product link (see [target](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#target)).
+   * This property is only leveraged when `display` is `grid`.
+   * @defaultValue `_self`
+   */
+  @Prop() gridCellLinkTarget: ItemTarget = '_self';
+
+  /**
+   * Sets a rendering function to bypass the standard HTML template mechanism for rendering products.
+   * You can use this function while working with web frameworks that don't use plain HTML syntax, e.g., React, Angular or Vue.
+   *
+   * Do not use this method if you integrate Atomic in a plain HTML deployment.
+   *
+   * @param productRenderingFunction
+   */
+  @Method() public async setRenderFunction(
+    productRenderingFunction: ItemRenderingFunction
+  ) {
+    this.itemRenderingFunction = productRenderingFunction;
+  }
```

Comment:
I assumed we want this feature for atomic-commerce too

Reply:
Yes this will be needed down the line 👍